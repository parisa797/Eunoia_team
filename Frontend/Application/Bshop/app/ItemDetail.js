import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  Image,
  StyleSheet,
  ImageBackground,
  TouchableOpacity,
  TouchableNativeFeedback,
  Platform,
  FlatList,
  Pressable,
  ScrollView,
} from "react-native";
import Ionicon from "react-native-vector-icons/Ionicons";
import Icon from "react-native-vector-icons/FontAwesome";
import AntIcon from "react-native-vector-icons/AntDesign";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import StarRating from "react-native-star-rating";
import Item from "./item";

const ItemDetail = ({ route, navigation }) => {
  const [liked, setLiked] = useState(false);
  console.log(route.params);

  var image = !route.params.photo ? null : route.params.photo;
  var price = `${route.params.price} تومان`;
  var discounted = `با تخفیف: ${route.params.price_with_discount} تومان`;
  var count =
    route.params.count != 0 ? `موجودی: ${route.params.count} عدد` : "ناموجود";
  var brand = !route.params.brand ? "برند نامشخص" : route.params.brand;
  const categories = {
    "Spices and condiments and food side dishes": "ادویه، چاشنی و مخلفات غذا",
    Cosmetics: "بهداشت و مراقبت پوست",
    "Makeup and trimming": "آرایش و پیرایش",
    Protein: "پروتئینی",
    "Junk Food": "تنقلات",
    Nuts: "خشکبار",
    "Sweets and desserts": "شیرینیجات و دسرها",
    perfume: "عطر، ادکلن و اسپری",
    "Fruits and vegetables": "غذا، کنسرو و سبزیجات",
    Dairy: "لبنیات",
    Drinks: "نوشیدنیها",
    "Washing and Cleaning Equipment": "وسایل شستشو و نظافت",
    others: "متفرقه",
  };
  var category = `دسته بندی: ${categories[route.params.category]}`;
  var manufacture = `تاریخ تولید: ${route.params.manufacture_jalali}`;
  var expire = `تاریخ انقضا: ${route.params.Expiration_jalali}`;

  return (
    <ScrollView nestedScrollEnabled={true} style={styles.container}>
      {/* <ImageBackground
        source={require("../assets/lemon.jpg")}
        style={styles.imagebackk}
      > */}
      {/* <View style={styles.shop}> */}

      <View style={styles.imageContainer}>
        <Pressable onPress={() => setLiked((isLiked) => !isLiked)}>
          <MaterialCommunityIcons
            name={liked ? "heart" : "heart-outline"}
            size={32}
            color={liked ? "red" : "black"}
          />
        </Pressable>
        {/* <Image style={styles.image} /> */}
        {image == null && (
          <Image
            resizeMode="contain"
            style={styles.image}
            source={require("../assets/no-image.png")}
          />
        )}
        {image && (
          <Image
            resizeMode="contain"
            style={styles.image}
            source={{ uri: "http://eunoia-bshop.ir:8000" + image }}
          />
        )}
      </View>
      <View style={styles.shop}>
        <View style={styles.details}>
          <Text style={styles.title}>{route.params.name} </Text>
          {/* <View style={styles.icon}>
              
              <Text style={styles.address}></Text>
            </View> */}
          <Text style={styles.title}>{route.params.description}</Text>
          <Text style={styles.title}>{brand}</Text>
          <Text style={styles.title}>{manufacture}</Text>
          <Text style={styles.title}>{expire}</Text>
          <Text style={styles.title}>{price} </Text>
          {route.params.price != route.params.price_with_discount && (
            <Text style={styles.title}>{discounted} </Text>
          )}
          <Text style={styles.title}>{category} </Text>
          <Text style={styles.title}>{count}</Text>
          <View>
            <StarRating
              starSize={25}
              disabled={true}
              fullStarColor={"#b31414"}
              rating={
                route.params.rate_value == 0 ? 0 : route.params.rate_value
              }
            ></StarRating>
          </View>
        </View>
      </View>
      <TouchableOpacity
        style={styles.SignUpBtn}
        // onPress={() => navigation.navigate("Comment", route.params.id)}
      >
        <Text style={styles.SignUpText}> مشاهده نظرات این محصول</Text>
      </TouchableOpacity>
      {/* </View> */}

      <Text style={{ fontWeight: "bold" }}></Text>

      {/* </ImageBackground> */}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  SignUpBtn: {
    borderRadius: 10,
    height: 40,
    width: 370,
    marginLeft: 12,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 15,
    marginBottom: 20,
    backgroundColor: "#b31414",
  },
  SignUpText: {
    color: "white",
    fontSize: 20,
    marginTop: 5,
    textAlign: "center",
  },

  icon: {
    justifyContent: "space-between",
    flexDirection: "row-reverse",
    alignItems: "baseline",
  },
  hozuri: { fontSize: 20, color: "red" },
  online: { fontSize: 20, color: "green" },
  shop: {
    borderRadius: 10,
    marginTop: 1,
    borderRadius: 10,
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 0, height: 2 },
    elevation: 5,
    backgroundColor: "white",
    height: 380,
    width: 370,
    marginLeft: 12,
    marginTop: 15,
  },
  rows: {
    borderRadius: 10,
    marginTop: 15,
    borderRadius: 10,
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 0, height: 2 },
    elevation: 5,
    backgroundColor: "#b31414",
    height: 40,
    width: 370,
    marginLeft: 12,
    fontWeight: "bold",
    fontSize: 25,
  },
  rowstext: {
    color: "white",
    fontSize: 20,
    marginTop: 5,
    textAlign: "center",
  },
  imagebackk: {
    width: 400,
    height: 700,
    marginTop: -20,
    marginLeft: -7,
  },
  items: {
    backgroundColor: "white",
    height: 210,
    width: 165,
    borderRadius: 10,
    marginTop: 15,
    borderRadius: 10,
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 1, height: 3 },
    elevation: 5,
    backgroundColor: "#f1f1f2",
    marginLeft: 12,
  },
  imageContainer: {
    backgroundColor: "white",
    // objectFit: ""
    width: "95%",
    height: 300,
    // alignItems: "center",
    // justifyContent: "center",
    // alignContent: "center",
    shadowColor: "black",
    shadowOpacity: 0.26,
    shadowOffset: { width: 0, height: 2 },
    elevation: 5,
    // borderTopLeftRadius: 10,
    // borderTopRightRadius: 10,
    borderRadius: 10,
    marginLeft: 11,
    marginTop: 8,

    // overflow: "hidden",
  },
  image: {
    // backgroundColor: "#f1f1f2",
    width: "80%",
    height: "80%",
    marginTop: "12%",
    resizeMode: "cover",
    alignItems: "center",
    justifyContent: "center",
    // paddingBottom: "1",
    marginLeft: 12,
  },
  details: {
    alignItems: "center",
    height: "48%",
    padding: 10,
  },
  title: {
    // fontFamily: "open-sans-bold",
    fontSize: 20,
    marginVertical: 5,
    // paddingTop: 3,
    marginLeft: "-1%",
    marginTop: "1%",
  },
  online_icon: {
    justifyContent: "space-between",
    flexDirection: "row-reverse",
    backgroundColor: "transparent",
    alignItems: "baseline",
  },
  address: {
    // fontFamily: "open-sans",
    fontSize: 20,
    color: "black",
  },
  mantaghee: {
    fontSize: 21,
    color: "black",
    marginLeft: 20,
  },
  container: {
    flex: 1,
    // backgroundColor: "#fff",
    // alignItems: "center",
    // justifyContent: "center",
    marginTop: "5%",
    // width: "50%",
    // height: "50%",
    marginBottom: "5%",
  },
});
export default ItemDetail;
