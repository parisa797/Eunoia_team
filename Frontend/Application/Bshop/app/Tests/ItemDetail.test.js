import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitFor,
  act,
} from "react-native-testing-library";
import renderer from "react-test-renderer";

import ItemDetail from "../ItemDetail";
// import * as Focus from "@react-navigation/native";

// const mockedFocused = jest.fn();
// jest.mock("@react-navigation/native", () => ({
//   ...jest.requireActual("@react-navigation/native"),
//   useIsFocused: () => mockedFocused,
// }));

describe("ItemDetail be tested", () => {
  // jest.mock("@react-navigation/native");
  const params = {
    rate_value: 2,
    photo: "kasd",
    price: 300,
    price_with_discount: 349,
    brand: null,
    category: "Protein",
    manufacture_jalali: "1400-04-12",
    Expiration_jalali: "1400-04-16",
    shop_id: 3,
    id: 0,
    name: "pofak",
    description: "sth",
  };

  it("renders correctly", () => {
    const push = jest.fn();

    const tree = renderer
      .create(<ItemDetail route={{ params }} navigation={{ push }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("test components inside itemdetail", async () => {
    const push = jest.fn();
    const { getByTestId, queryByTestId } = await render(
      <ItemDetail navigation={{ push }} route={{ params }} />
    );
    await new Promise((resolve) => setImmediate(resolve));

    var i = params.id;
    var name = [params.name, " "];
    var image = params.photo
      ? params.photo.includes("http://eunoia-bshop.ir:8000")
        ? params.photo
        : "http://eunoia-bshop.ir:8000" + params.photo
      : null;

    var price = `${params.price} ریال`;
    var discounted = `با تخفیف: ${params.price_with_discount} ریال`;
    var count = params.count != 0 ? `موجودی: ${params.count} عدد` : "ناموجود";
    var brand = !params.brand ? "برند نامشخص" : params.brand;
    const categories = {
      "Spices and condiments and food side dishes": "ادویه، چاشنی و مخلفات غذا",
      Cosmetics: "بهداشت و مراقبت پوست",
      "Makeup and trimming": "آرایش و پیرایش",
      Protein: "پروتئینی",
      "Junk Food": "تنقلات",
      Nuts: "خشکبار",
      "Sweets and desserts": "شیرینیجات و دسرها",
      perfume: "عطر، ادکلن و اسپری",
      "Fruits and vegetables": "غذا، کنسرو و سبزیجات",
      Dairy: "لبنیات",
      Drinks: "نوشیدنیها",
      "Washing and Cleaning Equipment": "وسایل شستشو و نظافت",
      others: "متفرقه",
    };
    var category = `دسته بندی: ${categories[params.category]}`;

    var manufacture = `تاریخ تولید: 12 تیر 1400`;
    var expire = `تاریخ انقضا: 16 تیر 1400`;

    // check item photo
    if (params.photo) {
      const photo = params.photo.includes("http://eunoia-bshop.ir:8000")
        ? params.photo
        : "http://eunoia-bshop.ir:8000" + params.photo;

      expect(getByTestId("item-detail-image-" + i).props.source.uri).toBe(
        photo
      );
      expect(queryByTestId("item-detail-noimage-" + i)).toBeNull();
    } else {
      expect(queryByTestId("item-detail-image-" + i)).toBeNull();
      expect(queryByTestId("item-detail-noimage-" + i)).not.toBeNull();
    }
    expect(getByTestId("item-detail-name-" + i).props.children).toStrictEqual(
      name
    );
    expect(getByTestId("item-detail-description-" + i).props.children).toBe(
      params.description
    );

    expect(getByTestId("item-detail-brand-" + i).props.children).toBe(brand);
    expect(getByTestId("item-detail-manufacture-" + i).props.children).toBe(
      manufacture
    );
    expect(getByTestId("item-detail-expire-" + i).props.children).toBe(expire);
    expect(getByTestId("item-detail-price-" + i).props.children).toStrictEqual([
      price,
      " ",
    ]);
    if (params.price != params.price_with_discount) {
      expect(
        getByTestId("item-detail-discounted-" + i).props.children
      ).toStrictEqual([discounted, " "]);
    } else {
      expect(queryByTestId("item-detail-discounted-" + i)).toBeNull();
    }

    expect(
      getByTestId("item-detail-category-" + i).props.children
    ).toStrictEqual([category, " "]);

    expect(getByTestId("item-detail-count-" + i).props.children).toBe(count);
  });
});
