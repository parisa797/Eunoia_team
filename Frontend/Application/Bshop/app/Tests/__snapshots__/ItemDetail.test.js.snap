// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail be tested renders correctly 1`] = `
<RCTScrollView
  nestedScrollEnabled={true}
  style={
    Object {
      "flex": 1,
      "marginBottom": "5%",
      "marginTop": "5%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 5,
          "height": 350,
          "marginLeft": 11,
          "marginTop": 8,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.26,
          "width": "95%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "http://eunoia-bshop.ir:8000kasd",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "height": "85%",
            "justifyContent": "center",
            "marginLeft": 35,
            "marginTop": "-1%",
            "resizeMode": "cover",
            "width": "80%",
          }
        }
        testID="item-detail-image-0"
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 5,
          "height": 380,
          "marginLeft": 12,
          "marginTop": 15,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.26,
          "width": 370,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": "48%",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-name-0"
        >
          pofak
           
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-description-0"
        >
          sth
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-brand-0"
        >
          برند نامشخص
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-manufacture-0"
        >
          تاریخ تولید: 12 تیر 1400
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-expire-0"
        >
          تاریخ انقضا: 16 تیر 1400
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-price-0"
        >
          300 ریال
           
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-discounted-0"
        >
          با تخفیف: 349 ریال
           
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-category-0"
        >
          دسته بندی: پروتئینی
           
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginLeft": "-1%",
              "marginTop": "1%",
              "marginVertical": 5,
            }
          }
          testID="item-detail-count-0"
        >
          موجودی: undefined عدد
        </Text>
        <View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              forwardedRef={[Function]}
              style={Object {}}
            >
              <View
                accessibilityRole="button"
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#b31414",
          "borderRadius": 10,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginLeft": 12,
          "marginTop": 15,
          "opacity": 1,
          "width": 370,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "marginTop": 5,
            "textAlign": "center",
          }
        }
      >
         مشاهده نظرات این محصول
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
  </View>
</RCTScrollView>
`;
